### PR TITLE
Add homebrew testing to jenkins

### PR DESCRIPTION
**Pull request by stanley-cheung**
For issue #1883

### CODE REVIEW

File: tools/jenkins/run_linuxbrew.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# This script is invoked by Jenkins and triggers a test run based on
```
Comment (jtattermusch) [inline tools/jenkins/run_linuxbrew.sh]: Update the comment

File: tools/jenkins/grpc_linuxbrew/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  bzip2 curl g++ gawk git m4 make patch ruby tcl wget
+
+# Install linuxbrew
+WORKDIR /home/linuxbrew
+ENV PATH /home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:$PATH
+ENV SHELL /bin/bash
+RUN git clone https://github.com/Homebrew/linuxbrew.git /home/linuxbrew/.linuxbrew
+RUN brew doctor || true
+
+# Install gRPC via linuxbrew
```
Comment (jtattermusch) [inline tools/jenkins/grpc_linuxbrew/Dockerfile]: I'd be leaning towards making the Dockerfile only contain the prerequisites and no perform the actual gRPC installation itself (which is what we want to test).

After the dockerfile is build (with homebrew and prereqs installed), you can do docker run that image with with appropriate command to test if homebrew installation works fine.

File: tools/jenkins/run_linuxbrew.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# env variable settings.
+set -ex
+
+sha1=$(sha1sum tools/jenkins/grpc_linuxbrew/Dockerfile | cut -f1 -d\ )
+DOCKER_IMAGE_NAME=grpc_linuxbrew_$sha1
+
+docker build -t $DOCKER_IMAGE_NAME tools/jenkins/grpc_linuxbrew \
+  >> report.xml || DOCKER_FAILED="true"
+
```
Comment (jtattermusch) [inline tools/jenkins/run_linuxbrew.sh]: add "docker run ...." here  to actually test the installation. If you don't   run what you've built, you don't need the trick with the Dockerfile sha1

Comment (nicolasnoble) [inline tools/jenkins/run_linuxbrew.sh]: The `>> report.xml` part seems nonsensical. What's it trying to do ? AFAIK, `docker build` isn't going to output any XML fragment to append to anything.

**Comment by jtattermusch** [comment]
Come on guys, just go to sleep, it's 1AM. :-)

**Comment by stanley-cheung** [comment]
Thanks for the feedback guys :) Will update the PR.

File: tools/jenkins/run_linuxbrew.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+set -ex
+
+sha1=$(sha1sum tools/jenkins/grpc_linuxbrew/Dockerfile | cut -f1 -d\ )
+DOCKER_IMAGE_NAME=grpc_linuxbrew_$sha1
+
+docker build -t $DOCKER_IMAGE_NAME tools/jenkins/grpc_linuxbrew
+
+supported="python nodejs ruby php"
+
+if [ "$language" == "core" ]; then
```
Comment (jtattermusch) [inline tools/jenkins/run_linuxbrew.sh]: on jenkins, then language is called "c"
https://grpc-testing.appspot.com/job/gRPC_master/

File: tools/jenkins/run_linuxbrew.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+# This script is invoked by Jenkins and triggers a test run of
+# linuxbrew installation of a selected language
+set -ex
+
+sha1=$(sha1sum tools/jenkins/grpc_linuxbrew/Dockerfile | cut -f1 -d\ )
+DOCKER_IMAGE_NAME=grpc_linuxbrew_$sha1
+
+docker build -t $DOCKER_IMAGE_NAME tools/jenkins/grpc_linuxbrew
+
+supported="python nodejs ruby php"
```
Comment (jtattermusch) [inline tools/jenkins/run_linuxbrew.sh]: list of what's supported will be determined by a condition in jenkins job settings, so there's probably not need to check explicitly here.

File: tools/jenkins/run_linuxbrew.sh (hunk 4 above)
Comment (jtattermusch) [inline tools/jenkins/run_linuxbrew.sh]: hm, I take it back, I will setup the jenkins job to use these constants for now.

**Comment by jtattermusch** [comment]
Jenkins failures are clearly unrelated, merging this.

**Pull request merged**